selfservice/flow/settings/handler.go

```diff
@@ -413,6 +413,22 @@ func (h *Handler) submitSettingsFlow(w http.ResponseWriter, r *http.Request, ps
 		break
 	}
 
+	// if the submission was from a recovery session, we have to end the recovery session and
+	// reissue a new session which isn't a recovery session anymore
+	if ss.Recovery {
+		if err := h.d.SessionManager().PurgeFromRequest(r.Context(), w, r); err != nil {
+			h.d.SettingsFlowErrorHandler().WriteFlowError(w, r, node.DefaultGroup, f, ss.Identity, err)
+			return
+		}
+		newSession := session.NewActiveSession(ss.Identity, h.d.Config(r.Context()), time.Now())
+
+		// this could sit nicely behind a flag to leave the user signed out after the password change
```

Comment:
This could sit behind a config flag, for example `selfservice.flows.recovery.after.keep_logged_in: bool`.

Reply:
I don't think this is a security related issue. From a user perspective, the latter one is definitely more ergonomic. Perhaps it might be worth asking why is it so common to require reauthenticating after recovery? This can always be done via some sort of reverse proxy setup but it is pretty complicated and non-ergonomic for the developer.

Having a flag gives the developer an option to choose. Maybe they do have certain ciso requirements to follow. Not having the option either disqualifies a product or puts a lot of effort on development team. Considering that the construct of having to reauth after recovery seems to be a common one and adds at most a few lines of code (setting flag), why not? I'm not fixed on this as being signed in works for me just fine.